diff --git a/target/linux/aj_net.c b/target/linux/aj_net.c
index 099f9ca..a828b81 100644
--- a/target/linux/aj_net.c
+++ b/target/linux/aj_net.c
@@ -781,6 +781,18 @@ static uint32_t chooseMDnsRecvAddr()
     struct ifaddrs* addrs;
     struct ifaddrs* addr;
 
+#ifndef NDEBUG
+    const char *env = getenv("ER_DEBUG_MDNS_RECV_ADDR");
+    if (env) {
+        struct in_addr ip;
+        if (inet_pton(AF_INET, env, &ip)) {
+            return ip.s_addr;
+        }
+
+        AJ_ErrPrintf(("chooseMDnsRecvAddr(): ER_DEBUG_MDNS_RECV_ADDR address (%s) is not a valid IPv4 address \n", env));
+    }
+#endif
+
     getifaddrs(&addrs);
     addr = addrs;
     while (addr != NULL) {
